hazelcast/src/test/java/com/hazelcast/kubernetes/KubernetesClientTest.java

```diff
@@ -751,6 +756,55 @@ public void endpointsFailFastWhenNoPublicAccess() {
         // exception
     }
 
+    @Test
+    public void apiAccessWithTokenRefresh() throws IOException {
+        // Token is read from the file, first token value is value-1
+        Path file = Files.createTempFile("token", null);
+        Files.write(file, "value-1".getBytes(StandardCharsets.UTF_8), StandardOpenOption.TRUNCATE_EXISTING);
+
+        stubFor(get(urlMatching("/apis/.*")).atPriority(1)
+                .withHeader("Authorization", equalTo("Bearer value-1"))
+                .willReturn(aResponse().withStatus(200).withBody("{}")));
+        stubFor(get(urlMatching("/api/.*")).atPriority(1)
+                .withHeader("Authorization", equalTo("Bearer value-1"))
+                .willReturn(aResponse().withStatus(200).withBody("{}")));
+        stubFor(get(urlMatching("/api/.*")).atPriority(1)
+                .withHeader("Authorization", equalTo("Bearer value-2"))
+                .willReturn(aResponse().withStatus(402).withBody("{}")));
+        stubFor(get(urlMatching("/apis/.*")).atPriority(1)
+                .withHeader("Authorization", equalTo("Bearer value-2"))
+                .willReturn(aResponse().withStatus(402).withBody("{}")));
+
+        KubernetesClient client = newKubernetesClient(new FileReaderTokenProvider(file.toString()));
+        client.endpoints();
+        assertFalse(client.isKnownExceptionAlreadyLogged());
+
+        // Token is rotated
+        Files.write(file, "value-2".getBytes(StandardCharsets.UTF_8), StandardOpenOption.TRUNCATE_EXISTING);
+
+        // Api server will not accept token with old value
+        stubFor(get(urlMatching("/apis/.*")).atPriority(1)
+                .withHeader("Authorization", equalTo("Bearer value-1"))
+                .willReturn(aResponse().withStatus(402).withBody("{}")));
+        stubFor(get(urlMatching("/api/.*")).atPriority(1)
+                .withHeader("Authorization", equalTo("Bearer value-1"))
+                .willReturn(aResponse().withStatus(402).withBody("{}")));
+        stubFor(get(urlMatching("/api/.*")).atPriority(1)
+                .withHeader("Authorization", equalTo("Bearer value-2"))
+                .willReturn(aResponse().withStatus(200).withBody("{}")));
+        stubFor(get(urlMatching("/apis/.*")).atPriority(1)
+                .withHeader("Authorization", equalTo("Bearer value-2"))
+                .willReturn(aResponse().withStatus(200).withBody("{}")));
+
+        client.endpoints();
+        assertFalse(client.isKnownExceptionAlreadyLogged());
+        try {
+            Files.delete(file);
```

Comment:
The file won't be deleted if the tests fail. Can we use [TemporaryFolder](https://junit.org/junit4/javadoc/4.12/org/junit/rules/TemporaryFolder.html) instead?